src/vs/workbench/contrib/testing/browser/testExplorerActions.ts

```diff
@@ -983,11 +983,15 @@ abstract class ExecuteTestAtCursor extends Action2 {
 	public async run(accessor: ServicesAccessor) {
 		const editorService = accessor.get(IEditorService);
 		const activeEditorPane = editorService.activeEditorPane;
-		const activeControl = editorService.activeTextEditorControl;
+		let activeControl = editorService.activeTextEditorControl;
 		if (!activeEditorPane || !activeControl) {
 			return;
 		}
 
+		if (isDiffEditor(activeControl)) {
```

Comment:
This PR assumes that the modified editor in the diff editor is always the one where tests are present, but this may not be the case.

I think we may want to use`ICodeEditorService.getActiveCodeEditor()` instead. The thing to check for there is whether the editor is embedded, via `if (editor instanceof EmbeddedCodeEditorWidget) { editor = editor.getParentEditor() }`

Reply:
Done. Removed checks for isCodeEditor given the change now uses activeCodeEditor.